pkg/configprofiles/provider.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package configprofiles
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/server/autoconfig/acprovider"
+	"github.com/cockroachdb/cockroach/pkg/server/autoconfig/autoconfigpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+)
+
+// profileTaskProvider is an implementation of acprovider.Provider
+// which reports task to execute from a static configuration profile.
+type profileTaskProvider struct {
+	syncutil.Mutex
+
+	envID       autoconfigpb.EnvironmentID
+	tasks       []autoconfigpb.Task
+	nextTaskIdx int
+}
+
+var _ acprovider.Provider = (*profileTaskProvider)(nil)
+
+// EnvUpdate is part of the acprovider.Provider interface.
+func (p *profileTaskProvider) EnvUpdate() <-chan struct{} {
```

Comment:
Might be worth noting the obvious: We return a channel that won't see updates because static profiles can't be updated.